File: backend/src/main/java/com/apocalypse/thefall/model/DiscoveryData.java
package com.apocalypse.thefall.model;

import com.apocalypse.thefall.entity.Tile;
import com.apocalypse.thefall.entity.character.Character;

import java.util.List;
import java.util.Set;

public record DiscoveryData(
        Character character,
        Set<Long> discoveredTileIds,
        List<Tile> allTiles,
        List<Tile> toDiscover
) {
}

File: backend/src/main/java/com/apocalypse/thefall/model/DiscoveryData.kt
package com.apocalypse.thefall.model

import com.apocalypse.thefall.entity.Tile
import com.apocalypse.thefall.entity.character.Character

@JvmRecord
data class DiscoveryData(
    val character: Character,
    val discoveredTileIds: Set<Long>,
    val allTiles: List<Tile>,
    val toDiscover: List<Tile>
)
